File: Scripts/newTask/factorial/java/Factorial.java
public class Factorial {

public static void main(String[] args) {
	
	for (int i = 0; i < 1000000; ++i)
		fact(1000);
}

public static long fact(final int n) {
    if (n < 0){
        System.err.println("No negative numbers");
        return 0;
    }
    return (n < 2) ? 1 : n * fact(n - 1);
}

}

File: Scripts/newTask/factorial/php/factorial.php
<?php
function factorial($n) {
  if ($n < 0) {
    return 0;
  }

  if ($n == 0) {
    return 1;
  }

  else {
    return $n * factorial($n-1);
  }
}

for ( $i = 0; $i < 1000000; $i++) {
 factorial(1000);
}

?>
